src/core/recv_buffer.c

```diff
@@ -23,23 +21,21 @@
     the queued up buffer.
 
     When physical buffer space runs out, assuming more 'virtual' space is
-    available, the physical buffer will be reallocated and copied over.
+    available, the physical buffer will be reallocated and may be copied over.
     Physical buffer space always doubles in size as it grows.
 
     The VirtualBufferLength is what is used to report the maximum allowed
     stream offset to the peer. Again, if the application drains at a fast
     enough rate compared to the incoming data, then this value can be much
     larger than the physical buffer. This has the effect of being able to
-    receive a large buffer (given a flight of packets) but not needed to
+    receive a large buffer (given a flight of packets) but not need to
     allocate memory for the entire buffer all at once.
 
     This does expose an attack surface though. In the common case we might be
     able to get by with a smaller buffer, but we need to be careful not to over
     commit. We must always be willing/able to allocate the buffer length
     advertised to the peer.
 
-    Currently, only growing the virtual buffer length is supported.
```

Comment:
AFAICT, this statement is still true. Why remove it?

Reply:
With app-owned mode, it is not strictly true anymore. The virtual buffer length decreases as we consume buffer space since we can't reuse it (and this is fine as long as it doesn't reduce the max byte index acceptable).